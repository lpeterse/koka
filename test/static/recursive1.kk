// recursive type
type list<a> {
    Nil
    Cons(head: a, tail: list<a>)
}
