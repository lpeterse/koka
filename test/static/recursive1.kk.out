static/recursive1/list: V -> V
Cons : forall<a> (head : a, tail : list<a>) -> list<a>
Nil : forall<a> list<a>
head : forall<a> (list : list<a>) -> exn a
isCons : forall<a> (list : list<a>) -> bool
isNil : forall<a> (list : list<a>) -> bool
tail : forall<a> (list : list<a>) -> exn list<a>